eng/common/scripts/Validate-Package.ps1

```diff
@@ -0,0 +1,312 @@
+  }
+}
+else {
+  # Login using PAT
+  LoginToAzureDevops $Devops_pat
+}
+
+az extension show -n azure-devops *> $null
+if (!$?){
+  az extension add --name azure-devops
+} else {
+  # Force update the extension to the latest version if it was already installed
+  # this is needed to ensure we have the authentication issue fixed from earlier versions
+  az extension update -n azure-devops *> $null
+}
+
+CheckDevOpsAccess
+
+class ValidationStatus
+{
+    [string]$Name
+    [string]$Status
+    [string]$Message
+
+    ValidationStatus([string]$name, [string]$status)
+    {
+        $this.Name = $name
+        $this.Status = $status
+    }
+}
+
+class PackageDetails
+{
+    [string]$Name
+    [string]$Version
+    [ValidationStatus]$VersionValidation
+    [ValidationStatus]$ChangeLogValidation
+    [ValidationStatus]$APIReviewValidation
+    [ValidationStatus]$PackageNameValidation
+
+    PackageDetails([string]$name)
+    {
+        $this.Name = $name
+    }
+}
+
+function ValidateVersion($packageName, $versionString)
+{
+    $validationStatus = [ValidationStatus]::new("Version Validation", "Success")
+    $version = [AzureEngSemanticVersion]::ParseVersionString($versionString)
+    if ($version -eq $null) {
+        $validationStatus.Status = "Failed"
+        $validationStatus.Message = "Version info is not available for package $packageName, because version '$versionString' is invalid. Please check if the version follows Azure SDK package versioning guidelines."    
+    }
+    return $validationStatus
+}
+
+function ValidateChangeLog($changeLogPath, $versionString)
+{
+    $validationStatus = [ValidationStatus]::new("Change Log Validation", "Success")
+    try
+    {
+        $changeLogFullPath = Join-Path $RepoRoot $changeLogPath
+        Write-Host "Path to change log: [$changeLogFullPath]"        
+        if (Test-Path $changeLogFullPath)
+        {
+            $errOutput = $( $validChangeLog = & Confirm-ChangeLogEntry -ChangeLogLocation $changeLogFullPath -VersionString $versionString -ForRelease $true ) 2>&1
+            if (!$validChangeLog) {
+                $validationStatus.Status = "Failed"
+                $validationStatus.Message = $errOutput
+            }
+        }
+        else {
+            $validationStatus.Status = "Failed"
+            $validationStatus.Message = "Change log is not found in [$changeLogPath]. Change log file must be present in package root directory."
+        }
+    }
+    catch
+    {
+        Write-Host "Current directory: $(Get-Location)"
+        $validationStatus.Status = "Failed"
+        $validationStatus.Message = $_.Exception.Message
+    }    
+    return $validationStatus
+}
+
+function VerifyAPIReview($packageDetails, $packageName, $packageVersion, $language)
+{
+    $packageDetails.APIReviewValidation = [ValidationStatus]::new("API Review Approval", "Pending")
+    $packageDetails.PackageNameValidation = [ValidationStatus]::new("Package Name Approval", "Pending")
+
+    try
```

Comment:
Should we be sharing this logic with other places?

Reply:
Current Prepare release script already uses this API. I will update it to pass package version param. Create API review gets the status of matching revision, so this API is not really required in that scenario to check the status.